src/main/java/com/hackerrank/weather/controller/WeatherApiRestController.java

```diff
@@ -42,7 +42,7 @@ public List<Weather> getWeatherByCityName(@PathVariable("name") String cityName)
         return weatherService.getWeatherByCityName(cityName);
     }
 
-    @PostMapping
+    @PostMapping("/weather")
```

Comment:
Clean up that `/weather` as it results in path `/weather/weather which is wrong.